### PR TITLE
Extracting progress handling to a class

### PR DESCRIPTION
**Pull request by ferdonline**
Some refactoring to `sim/fadvance_core.cpp`, gathering code related to update progress into its own class.

### CODE REVIEW

**Review by alkino — APPROVED** [review]
Nice ;) !

**Comment by ferdonline** [comment]
Anything blocking merge?

**Review by iomaganaris — APPROVED** [review]

File: coreneuron/sim/fadvance_core.cpp [hunk 1]
```diff
@@ -30,6 +30,45 @@ extern corenrn_parameters corenrn_param;
 static void* nrn_fixed_step_thread(NrnThread*);
 static void* nrn_fixed_step_group_thread(NrnThread*, int, int, int&);
 
+
+namespace {
+
+class ProgressBar {
+    bool show;
```
Comment (olupton) [inline coreneuron/sim/fadvance_core.cpp:37]: Nitpick: better to put this after `pbar` and `current_step` to keep `sizeof(ProgressBar)` smaller.

File: coreneuron/sim/fadvance_core.cpp [hunk 2]
```diff
@@ -30,6 +30,45 @@ extern corenrn_parameters corenrn_param;
...
+        if (show && (current_step % progressbar_update_steps) == 0) {
+            progressbar_update(pbar, current_step, time);
+        }
+    }
+
+    void step(double time) {
+        update(current_step + 1, time);
+    }
+
+    virtual ~ProgressBar() {
```
Comment (olupton) [inline coreneuron/sim/fadvance_core.cpp:62]: Nitpick: why `virtual`?

**Review by olupton — APPROVED** [review]
LGTM, 2 tiny comments

File: coreneuron/sim/fadvance_core.cpp (hunk 1 above)
Comment (ferdonline) [reply coreneuron/sim/fadvance_core.cpp:37]: the lost art of struct packing :)

File: coreneuron/sim/fadvance_core.cpp (hunk 2 above)
Comment (ferdonline) [reply coreneuron/sim/fadvance_core.cpp:62]: Just because it's a good habit. But in this case maybe it's not useful.

**Pull request merged**